edweiss-app/components/todo/abstractTodoEditor.tsx

```diff
@@ -0,0 +1,314 @@
+/**
+ * @file abstractTodoEditor.tsx
+ * @description Module for editing todo items in the edweiss app
+ * @author Adamm Alaoui
+ */
+
+// ------------------------------------------------------------
+// --------------- Import Modules & Components ----------------
+// ------------------------------------------------------------
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import RouteHeader from '@/components/core/header/RouteHeader';
+import Icon from '@/components/core/Icon';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { callFunction } from '@/config/firebase';
+import t from '@/config/i18config';
+import { LightDarkProps } from '@/constants/Colors';
+import Todolist from '@/model/todo';
+import { Time } from '@/utils/time';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { router } from 'expo-router';
+import { useState } from 'react';
+import Toast from 'react-native-toast-message';
+import { sameTodos } from '../../utils/todo/utilsFunctions';
+import { StatusChanger } from './todoDisplay';
+import Functions = Todolist.Functions;
+
+// Types
+type TodoStatus = Todolist.TodoStatus;
```

Comment:
Your code are very long, it's a good idea to extract some componant to better readability.